app/lib/encryption/room.ts

```diff
@@ -268,17 +276,51 @@ export default class EncryptionRoom {
 					tmsg = await this.decryptText(tmsg);
 				}
 
-				return {
+				const decryptedMessage: IMessage = {
 					...message,
 					tmsg,
 					msg,
-					e2e: E2E_STATUS.DONE
+					e2e: 'done'
 				};
+
+				const decryptedMessageWithQuote = await this.parseQuoteAttachment(decryptedMessage);
+				return decryptedMessageWithQuote;
 			}
 		} catch {
 			// Do nothing
 		}
 
 		return message;
 	};
+
+	async parseQuoteAttachment(message: IMessage) {
+		const urls = message.msg?.match(getMessageUrlRegex()) || [];
+		await Promise.all(
+			urls.map(async (url: string) => {
+				const parsedUrl = parse(url, true);
+				const messageId = parsedUrl.query?.msg;
+				if (!messageId || Array.isArray(messageId)) {
+					return;
+				}
+
+				const messageFromDB = await getMessageByQuery(Q.and(Q.where('id', messageId), Q.where('e2e', E2E_STATUS.DONE)));
+				if (messageFromDB?.length) {
+					const decryptedQuoteMessage = mapMessageFromDB(messageFromDB[0]);
+					message.attachments = message.attachments || [];
+					const quoteAttachment = createQuoteAttachment(decryptedQuoteMessage, url);
+					return message.attachments.push(quoteAttachment);
+				}
+
+				const quotedMessageObject = await getSingleMessage(messageId);
+				if (!quotedMessageObject) {
+					return;
+				}
+				const decryptedQuoteMessage = await this.decrypt(mapMessageFromApi(quotedMessageObject));
+				message.attachments = message.attachments || [];
+				const quoteAttachment = createQuoteAttachment(decryptedQuoteMessage, url);
+				return message.attachments.push(quoteAttachment);
+			})
+		);
```

Comment:
I can see that are some lines repeating the same code, but I don't know if I change the `decryptedQuoteMessage` to let and use if/else as below it's more readable than that one
```suggestion
				let decryptedQuoteMessage: IMessage
				const messageFromDB = await getMessageByQuery(Q.and(Q.where('id', messageId), Q.where('e2e', E2E_STATUS.DONE)));
				if (messageFromDB?.length) {
					decryptedQuoteMessage = mapMessageFromDB(messageFromDB[0]);
				} else {
					const quotedMessageObject = await getSingleMessage(messageId);
					if (!quotedMessageObject) {
						return;
					}
					decryptedQuoteMessage = await this.decrypt(mapMessageFromApi(quotedMessageObject));
				}

				if(!decryptedQuoteMessage) {
					return
				}

				message.attachments = message.attachments || [];
				const quoteAttachment = createQuoteAttachment(decryptedQuoteMessage, url);
				return message.attachments.push(quoteAttachment);
			})
		);
```

Reply:
I prefer the way it's now with early returns.